### PR TITLE
Week 3 assignment

### PR DESCRIPTION
**Pull request by sachin286**

### CODE REVIEW

**Review by ShashankSuresh — CHANGES_REQUESTED** [review]
'Week-3/Exercise-3.2/apply or call.js' - The name of this file could be like this 'call-bind-apply.js' or 'callBindApply.js'. Its good to avoid spaces in file names.

**Review by ShashankSuresh — CHANGES_REQUESTED** [review]
Instead of having 2 files for call bind and apply example, we can have just 1 file which as all the 3 implementation.

**Review by ShashankSuresh — CHANGES_REQUESTED** [review]
'Week-3/Exercise-3.3/Output.txt' - Just to maintain consistency we have all files in lower case?

File: Week-3/Exercise-3.4/demo.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+function createStack() {
```
Comment (ShashankSuresh) [inline Week-3/Exercise-3.4/demo.js:1]: Great job.. :)

File: Week-3/Exercise-3.1/memoise.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+//create a memoise function
+
+function memoize(func) {
```
Comment (ShashankSuresh) [inline Week-3/Exercise-3.1/memoise.js:3]: Great job.. :)

**Review by ShashankSuresh — CHANGES_REQUESTED** [review]
Kindly do the necessary changes.

**Comment by sachin286** [comment]
@ShashankSuresh changes have been implemented

**Review by ShashankSuresh — APPROVED** [review]
Great work. :)

**Pull request merged**